### ISSUE TITLE
Feature proposal: intent score evaluation

### ISSUE THREAD

**Issue by zerowon01** [yoichiro/dialogflow-query-checker#20]
Thank you for this tool; very handy. While using it I got to thinking it would nice to know how well my intents were matching, score-wise. Either a verbose output and/or a score threshold for passing condition would be quite welcome.

**Comment by yoichiro** [comment]
@zerowon01 I agree your suggestion, thank you. I think the following specification to specify the score threshold value, and this tool will check its actual value, if the threshold value is specified in the yaml file.

```yaml
...
tests:
  -
    condition:
      ...
    expect:
      ...
      scoreThreshold: 0.98
```

Does the specification above sound good for you?

**Comment by zerowon01** [comment]
Thanks for considering this enhancement. That was what I had in mind as well. However, the one counter-proposal that I have, but am NOT entirely confident with is, INSTEAD of having a per-test scoreThreshold, to perhaps have a GLOBAL scoreThreshold. The goal / concern I have for/against the proposal is: 1) From the test writer it may be convenient to write less rules as it may be common to have a "global threshold" way of thinking  2) However, then you lose your per-test granularity and 3) I can imagine that the test logic implementation could become more (unnecessarily so?) complex. 
So to recap I think your proposal is the way forward.

**Comment by yoichiro** [comment]
I agree your idea. Actually, most developers should think one global threshold value, I guess, too. But, the developers may want to specify a different value for some test-cases. To achieve these two values, I would like to introduce a `defaultScoreThreshold` definition like the following:

```yaml
...
tests:
  defaultScoreThreshold: 0.98
  -
    condition:
      ...
    expect:
      ...
      scoreThreshold: 0.95
```

Basically, the `defaultScoreThreshold` value will be applied for all test-cases. Also, developers can override the value for each test-case using `scoreThreshold`. This means that the `scoreThreshold` takes a priority than the `defaultScoreThreshold`. And, behaviors at omitting each definition are the following:

* `defaultScoreThreshold` specified, `scoreThreshold` omitted - `defaultScoreThreshold` value is applied.
* `defaultScoreThreshold` omitted, `scoreThreshold` specified - `scoreThreshold` value is applied.
* `defaultScoreThreshold` omitted, `scoreThreshold` omitted - Not checked about the score value.

@zerowon01 How does that sound?

**Comment by zerowon01** [comment]
Thanks for the quick feedback. Covering both cases seems ideal! The three evaluations you explicitly mention below cover everything I can think of and/or would be interested in.

**Comment by yoichiro** [comment]
@zerowon01 Ok, good. :)

**Comment by zerowon01** [comment]
Forgot to mention earlier today but it sure be nice if the score showed up on the junior XML results. Can open a second request is necessary.

**Comment by yoichiro** [comment]
@zerowon01 Could you tell me more detail of the purpose by including the result in the JUnit XML results? Especially, I didn't understand your mention below:

> Can open a second request is necessary.

**Comment by yoichiro** [comment]
@zerowon01 The current output of a JUnit XML is like the following:

```xml
<testsuite name="dialogflow-query-checker" tests="52" errors="0" failures="0" time="6.259749">
  <testcase name="Event: GOOGLE_ASSISTANT_WELCOME" assertions="6" time="0.394515"/>
  <testcase name="Query: give me a rule (game)" assertions="6" time="0.284914"/>
  <testcase name="Query: Update the answer to 123 (game)" assertions="7" time="0.208070"/>
  <testcase name="Query: hint (game)" assertions="6" time="0.205718"/>
  ...
```

Do you think where should each score value be written? It would be great if you propose some example you expect.

**Comment by yoichiro** [comment]
Pull request #29 without writing score values to the JUnit XML.

**Comment by zerowon01** [comment]
I was thinking the score could be written for each testcase score="0.85"

**Comment by yoichiro** [comment]
@zerowon01 I have released a new version 1.15.0 which includes checking each score value and writing each score value to the JUnit XML file. I close this issue. Thank you for this proposal!
